### PR TITLE
py-aubio: fix build on macOS before Mavericks

### PR DESCRIPTION
**Pull request by catap**
###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS x.y
Xcode x.y

###### Verification 
Have you

- [ ] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [ ] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [ ] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

File: python/py-aubio/Portfile [hunk 1]
```diff
@@ -32,5 +32,13 @@ if {${name} ne ${subport}} {
     depends_lib-append \
                     port:py${python.version}-numpy
 
+    # macOS 10.8 (Mountain Lion, darwin 12) or below requires fftw-3
+    if {${os.major} < 13 } {
```
Comment (reneeotten) [inline python/py-aubio/Portfile:36]: please use here: `if {${os.platform} eq "darwin" && ${os.major} .....`

**Review by mascguy — APPROVED** [review]
Looks good overall, but I agree with the change suggested by @reneeotten.

**Comment by reneeotten** [comment]
that change looks good - thanks! Two more questions about the dependencies: (1) why did you add `py-pip` as a build dependency, that's typically not needed (I know, you didn't change that in this PR...) and (2) do we understand why the optional `fftw` dependency is actually required on earlier OSes?

**Comment by catap** [comment]
@reneeotten, (2) `fftw` is required for old macOS because before Apple's vDSP hasn't got some API and build is failed:
```
src/spectral/dct_accelerate.c:51:14: warning: implicit declaration of function 'vDSP_DCT_CreateSetup' is invalid in C99 [-Wimplicit-function-declaration]
  s->setup = vDSP_DCT_CreateSetup(NULL, (vDSP_Length)size, vDSP_DCT_II);
             ^
src/spectral/dct_accelerate.c:51:60: error: use of undeclared identifier 'vDSP_DCT_II'
  s->setup = vDSP_DCT_CreateSetup(NULL, (vDSP_Length)size, vDSP_DCT_II);
                                                           ^
src/spectral/dct_accelerate.c:52:63: error: use of undeclared identifier 'vDSP_DCT_III'; did you mean 'vDSP_DFT_zop'?
  s->setupInv = vDSP_DCT_CreateSetup(NULL, (vDSP_Length)size, vDSP_DCT_III);
                                                              ^~~~~~~~~~~~
                                                              vDSP_DFT_zop
/System/Library/Frameworks/vecLib.framework/Headers/vDSP.h:8029:6: note: 'vDSP_DFT_zop' declared here
void vDSP_DFT_zop(
     ^
src/spectral/dct_accelerate.c:52:15: warning: incompatible integer to pointer conversion assigning to 'vDSP_DFT_Setup' (aka 'struct vDSP_DFT_SetupStruct *') from 'int' [-Wint-conversion]
  s->setupInv = vDSP_DCT_CreateSetup(NULL, (vDSP_Length)size, vDSP_DCT_III);
              ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/spectral/dct_accelerate.c:74:3: warning: implicit declaration of function 'vDSP_DCT_Execute' is invalid in C99 [-Wimplicit-function-declaration]
  vDSP_DCT_Execute(s->setup, (const float *)input->data, (float *)output->data);
  ^
3 warnings and 2 errors generated.
```

(1) I guess it was a copy and paste issue. `py-pip` doesn't required here. Let me remove it.

**Review by mascguy — APPROVED** [review]
LGTM. @reneeotten?

**Comment by catap** [comment]
@mascguy can I ask you to merge also this one?

**Comment by mascguy** [comment]
> @mascguy can I ask you to merge also this one?

@reneeotten any final thoughts/concerns, before we merge this?

**Pull request merged**